### DISCUSSION TITLE
problem Run my TTS and Vocoder model

### DISCUSSION THREAD

**Discussion by Al-Fahrezi** [coqui-ai/TTS#2531] [General Q&A] [answered]
when i want to run to get output with vits model and wavernn model on google colab. there is an error

> Runtimeerror: given groups=1, weight of size [128, 80, 5], expected input[1, 213248, 5] to have 80 channels, but got 213248 channels instead

Vits  code:
`import os

from trainer import Trainer, TrainerArgs

from TTS.tts.configs.shared_configs import BaseDatasetConfig
from TTS.tts.configs.vits_config import VitsConfig
from TTS.tts.datasets import load_tts_samples
from TTS.tts.models.vits import Vits, VitsAudioConfig
from TTS.tts.utils.text.tokenizer import TTSTokenizer
from TTS.utils.audio import AudioProcessor

output_path = os.path.dirname("/content/drive/MyDrive/hasil")
dataset_config = BaseDatasetConfig(
    formatter="ljspeech", meta_file_train="LIST.txt", path=os.path.join("/content/drive/MyDrive/dataset")
)
audio_config = VitsAudioConfig(
    sample_rate=22050, win_length=1024, hop_length=256, num_mels=80, mel_fmin=0, mel_fmax=None
)

config = VitsConfig(
    audio=audio_config,
    run_name="vits_ljspeech",
    batch_size=32,
    eval_batch_size=16,
    batch_group_size=5,
    num_loader_workers=8,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=1000,
    text_cleaner="phoneme_cleaners",
    use_phonemes=True,
    phoneme_language="id",
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    compute_input_seq_cache=True,
    print_step=25,
    print_eval=True,
    mixed_precision=True,
    output_path=output_path,
    datasets=[dataset_config],
    cudnn_benchmark=False,
)
ap = AudioProcessor.init_from_config(config)

tokenizer, config = TTSTokenizer.init_from_config(config)

train_samples, eval_samples = load_tts_samples(
    dataset_config,
    eval_split=True,
    eval_split_max_size=config.eval_split_max_size,
    eval_split_size=config.eval_split_size,
)

model = Vits(config, ap, tokenizer, speaker_manager=None)

trainer = Trainer(
    TrainerArgs(continue_path = "/content/drive/MyDrive/vits_ljspeech-April-16-2023_05+28PM-0000000", skip_train_epoch=True),
    config,
    output_path,
    model=model,
    train_samples=train_samples,
    eval_samples=eval_samples,
)
trainer.fit()`

WaveRnn Code:
`import os

from trainer import Trainer, TrainerArgs

from TTS.utils.audio import AudioProcessor
from TTS.vocoder.configs import WavernnConfig
from TTS.vocoder.datasets.preprocess import load_wav_data
from TTS.vocoder.models.wavernn import Wavernn

output_path = os.path.dirname("/content/drive/MyDrive/hasil")
config = WavernnConfig(
    batch_size=64,
    eval_batch_size=16,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=10000,
    seq_len=1280,
    pad_short=2000,
    use_noise_augment=False,
    eval_split_size=10,
    print_step=25,
    print_eval=True,
    mixed_precision=False,
    lr=1e-4,
    grad_clip=4,
    data_path=os.path.dirname("/content/drive/MyDrive/dataset/wavs"),
    output_path=output_path,
)

ap = AudioProcessor(**config.audio.to_dict())

eval_samples, train_samples = load_wav_data(config.data_path, config.eval_split_size)

model = Wavernn(config)

trainer = Trainer(
    TrainerArgs(continue_path = "/content/drive/MyDrive/run-April-12-2023_09+11AM-0000000"),
    config,
    output_path,
    model=model,
    train_samples=train_samples,
    eval_samples=eval_samples,
    training_assets={"audio_processor": ap},
)
trainer.fit()`

Run my vits models and Vocoder models
`!tts --text "Maaf, sebagai model bahasa, saya tidak dapat membuat contoh bobot model pth karena itu tergantung pada arsitektur dan jenis model yang digunakan. Namun, jika Anda memiliki file model pth yang ingin Anda lihat bobotnya, Anda dapat menggunakan PyTorch untuk membuka file tersebut dan melihat struktur serta bobotnya" \
    --config_path /content/drive/MyDrive/vits_ljspeech-April-16-2023_05+28PM-0000000/config.json \
    --model_path /content/drive/MyDrive/vits_ljspeech-April-16-2023_05+28PM-0000000/best_model_7482.pth \
    --vocoder_path /content/drive/MyDrive/run-April-12-2023_09+11AM-0000000/best_model_7101.pth \
    --vocoder_config_path /content/drive/MyDrive/run-April-12-2023_09+11AM-0000000/config.json
    --out_path /content/drive/MyDrive/sound.wav
`
is the model not compatible?

Sorry, my English is bad

**Comment by p0p4k** [comment]
vits does not need vocoder, it is built inside of it.

**Reply by Al-Fahrezi** [reply]
Is that true? but the sound results from the model is still unclear. what's the solution?

**Reply by p0p4k** [reply, marked as answer]
Train longer or have a better dataset.
